plugins/cad/src/components/ResourceEditorDialog/components/PxeParametricEditor/nodes/widget/PxeRosterWidgetNode.tsx

```diff
@@ -0,0 +1,174 @@
+/**
+ * Copyright 2024 The Nephio Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Button, makeStyles } from '@material-ui/core';
+import AddIcon from '@material-ui/icons/Add';
+import DeleteIcon from '@material-ui/icons/Delete';
+import { get } from 'lodash';
+import React, { Fragment } from 'react';
+import { IconButton } from '../../../../../Controls';
+import { useEditorStyles } from '../../../FirstClassEditors/styles';
+import { PxeParametricEditorNodeProps } from '../../PxeParametricEditorNode';
+import {
+  PxeRosterWidgetEntry,
+  PxeValueType,
+} from '../../types/PxeConfiguration.types';
+import {
+  PxeResourceChangeRequest,
+  PxeValue,
+} from '../../types/PxeParametricEditor.types';
+import { createResourceChunkAfterChangeRequest } from '../../utils/createResourceChunkAfterChangeRequest';
+import { generateValueLabel } from '../../utils/generateLabelsForWidgets';
+import {
+  arrayWithItemRemoved,
+  arrayWithItemReplaced,
+} from '../../utils/general/immutableArrays';
+import { PxeParametricEditorNodeList } from '../../PxeParametricEditorNodeList';
+
+type RosterItemResourceChunk = {
+  readonly key: string;
+  readonly value: PxeValue;
+};
+
+type RosterValueType = PxeValueType.Object | PxeValueType.Array;
+
+// TODO Consider refactoring this component after Prettier settings update.
+export const PxeRosterWidgetNode: React.FC<PxeParametricEditorNodeProps> = ({
+  configurationEntry,
+  resourceChunk,
+  onResourceChangeRequest,
+}) => {
+  const { values, itemEntries } = configurationEntry as PxeRosterWidgetEntry;
```

Comment:
Is it possible not to use casting with `as` .

Reply:
It would be possible to add a check which would auto-infer the valid type, but the check would be somewhat redundant. Node resolution mechanism "guarantees" proper entry type when specific node is created.

Having said that, I guess that a guard clause (with mid-function `throw` 😉) wouldn't hurt and could attribute to early detection of resolution code error. For example, every specific node could start with something like:
```ts
if (configurationEntry.type !== PxeConfigurationEntryType.Foobar) {
  throw new Error('Invalid configuration entry for node Foobar.')
}
// here TS should know the specific type of configurationEntry
```

I might add this in future PR.